src/main/java/se/sundsvall/casedata/service/scheduler/emailreader/EmailReaderWorker.java

```diff
@@ -47,35 +48,46 @@ public EmailReaderWorker(final MessageRepository repository, final ErrandReposit
 		this.notificationService = notificationService;
 	}
 
-	void getAndProcessEmails() {
+	@Transactional
+	public void getAndProcessEmails() {
 
 		try {
 			emailReaderClient.getEmail(emailReaderProperties.municipalityId(), emailReaderProperties.namespace())
-				.forEach(this::saveAndRemoteDelete);
+				.forEach(email ->
+
+				{
+					if (saveAndRemoteDelete(email)) {
+						deleteMail(email);
```

Comment:
If the operation fails in the 5:th call of  a set of 10 emails?
Will not this rollback the transaction? Will not this 5 emails be lost?